src/components/HtmlInjection.tsx

```diff
@@ -1,21 +1,42 @@
 'use client';
 
 import type { HTMLAttributes, ScriptHTMLAttributes } from 'react';
-import { useEffect, useMemo } from 'react';
+import { useEffect, useMemo, useRef } from 'react';
+
+import { noop } from '../lib';
 
 interface Props extends HTMLAttributes<HTMLDivElement> {
     html: string;
     onError: () => void;
+    onPlay?: () => void;
 }
 
 const IFRAMELY_EMBED_SCRIPT_SRC = '//cdn.iframe.ly/embed.js';
 
 export function HtmlInjection(props: Props) {
-    const { html, onError, ...attrs } = props;
+    const { html, onError, onPlay = noop, ...attrs } = props;
+    const containerRef = useRef<HTMLDivElement>(null);
 
     const strippedHtml = useScripts(html, onError);
 
-    return <div {...attrs} dangerouslySetInnerHTML={{ __html: strippedHtml }} />;
+    useEffect(() => {
+        if (!containerRef.current) {
+            return;
+        }
+
+        import('player.js').then((playerjs) => {
```

Comment:
Since the player.js library relies on `window` which is not available in Node (i.e. during tests), I've opted for dynamic import to only import this package when the effect is triggered in the browser.

Due to this I also had to change the `module` option in TS config to `es2020`.